src/lib.rs

```diff
@@ -126,8 +139,81 @@ pub fn select_coin_knapsack(
 fn knap_sack(
     adjusted_target: u64,
     smaller_coins: &[(usize, OutputGroup)],
+    inputs: &[OutputGroup],
+    options: CoinSelectionOpt,
 ) -> Result<SelectionOutput, SelectionError> {
-    unimplemented!()
+    let mut target_reached = false;
+    let mut selected_inputs: HashSet<(usize)> = HashSet::new();
+    let mut accumulated_value: u64 = 0;
+    let mut accumulated_weight: u32 = 0;
+    let mut best_set: HashSet<(usize)> = HashSet::new();
+    // Assigning infinity to beginwith
+    let mut best_set_value: u64 = u64::MAX;
+
+    for i in 1..=1000 {
+        for j in 1..=2 {
+            if !target_reached {
+                for &(index, u) in smaller_coins {
+                    //Simulate a coin toss
+                    let mut rng = thread_rng();
+                    let prob = 0.5;
+                    let toss_result: bool = rng.gen_bool(prob);
```

Comment:
no need to create these variables.
using rng.gen_bool(0.5) in the if condition is more readable